### PR TITLE
Prod Comparison: Ads stage revert batching

### PR DESCRIPTION
**Pull request by Sneagan**
This PR exists only to demonstrate what merging #153 would cause the production PR to look like. This PR should not be merged.

### CODE REVIEW

File: server/db.go [hunk 1]
```diff
@@ -318,6 +318,12 @@ func (c *Server) fetchIssuers(issuerType string) (*[]Issuer, error) {
 }
 
 func (c *Server) FetchAllIssuers() (*[]Issuer, error) {
+	if c.caches != nil {
```
Comment (husobee) [inline server/db.go:321]: is the issuer lookup so slow we need caching, or is this an optimization?

File: server/dynamo.go [hunk 2]
```diff
@@ -33,15 +33,7 @@ func (c *Server) InitDynamo() {
...
-func (c *Server) fetchRedemptionV2(issuer *Issuer, ID string) (*RedemptionV2, error) {
-	issuerUUID, err := uuid.FromString(issuer.ID)
-	if err != nil {
-		c.Logger.Error("Bad issuer id")
-		return nil, errors.New("Bad issuer id")
-	}
-
-	id := uuid.NewV5(issuerUUID, ID)
-
+func (c *Server) fetchRedemptionV2(id uuid.UUID) (*RedemptionV2, error) {
```
Comment (husobee) [inline server/dynamo.go:36]: maybe a function comment stating that the caller should prepare the uuid identifier appropriately to the implementation? (uuidv5 in this case?)

File: utils/errors.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	return &ProcessingError{
+		Cause:          err,
+		FailureMessage: message,
+		Temporary:      temporary,
+		Backoff:        backoff,
+		KafkaMessage:   kafkaMessage,
+	}
+}
+
+func ErrorIsTemporary(err error, logger *zerolog.Logger) (bool, time.Duration) {
```
Comment (husobee) [inline utils/errors.go:44]: comment on exported function

File: utils/errors.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+// Error makes ProcessingError an error
+func (e ProcessingError) Error() string {
+	msg := fmt.Sprintf("error: %s", e.FailureMessage)
+	if e.Cause != nil {
+		msg = fmt.Sprintf("%s: %s", msg, e.Cause)
+	}
+	return msg
+}
+
+func ProcessingErrorFromErrorWithMessage(
```
Comment (husobee) [inline utils/errors.go:28]: comment on exported function

**Review by husobee — CHANGES_REQUESTED** [review]
Suggest linting corrections.

File: server/db.go (hunk 1 above)
Comment (Sneagan) [reply server/db.go:321]: This is an optimization because the issuers change rarely and this function gets called a lot.

**Pull request closed without merging**